server/src/gamestate.rs

```diff
@@ -43,10 +50,49 @@ impl GameState {
 
     /// Updates the game state in one tick.
     pub fn process_game_update(&mut self) {
+        // Do a normal position update
         for (_, player) in &mut self.players {
             player.x += player.move_x.unwrap_or(0.0);
             player.y += player.move_y.unwrap_or(0.0);
         }
+
+        let mut destroyed_bullets = Vec::new();
+        let mut destroyed_players = Vec::new();
+
+        for (_, bullet) in &mut self.bullets {
+            bullet.x += bullet.move_x.unwrap_or(0.0);
+            bullet.y += bullet.move_y.unwrap_or(0.0);
+
+            // Hardcoded map boundaries
+            if bullet.x < 0.0 || bullet.x > 500.0 || bullet.y < 0.0 || bullet.y > 500.0 {
+                destroyed_bullets.push(bullet.id);
+            }
+        }
+
+        // Check for collisions
+        for (_, bullet) in &mut self.bullets {
+            for (_, player) in &mut self.players {
+                let dx = bullet.x - player.x;
+                let dy = bullet.y - player.y;
+                let dist = (dx * dx + dy * dy).sqrt();
+                if dist < BULLET_RADIUS + PLAYER_RADIUS {
```

Comment:
I would agree with that, but I think I have heard plenty of other talk the they should be square. Make them circles, it's easier as you say, and size is a good enough distinction.